src/scripts/delete-user.test.ts

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+/* eslint-disable functional/no-return-void */
+/* eslint-disable functional/no-expression-statement */
+/* eslint-disable functional/functional-parameters */
+import * as TE from 'fp-ts/TaskEither';
+import * as O from 'fp-ts/Option';
+import { UserService } from './services/user-service';
+import { deleteUser } from './delete-user';
+
+import {
+  baseUserService,
+  user,
+  deactivatedUser,
+} from './__fixtures__/data-providers';
+
+describe('Deleting users without using force', () => {
+  it('passes when attempting to delete a deprovisioned user', async () => {
+    // Given a deleteUser function that can succsessfully delete a given user
+    const userService: UserService = {
+      ...baseUserService(),
+      deleteUser: () => TE.right(deactivatedUser),
+      getUser: () => TE.right(O.some(deactivatedUser)),
+    };
+
+    // When we attempt to delete an existing deprovisioned user
+    const result = await deleteUser(userService, deactivatedUser.id, false)();
+
+    // Then we should have a right
+    expect(result).toEqualRight(deactivatedUser);
```

Comment:
One last change; assert that `deleteUser` was called here.

Reply:
Asserted